--- mprsyncup.orig	2024-07-15 13:39:16.000000000 -0500
+++ mprsyncup	2024-07-16 13:04:34.000000000 -0500
@@ -31,15 +31,15 @@
 set -x
 
 # Paths we'll work on:
-ROOT=/var/tmp/macports
+ROOT=/Volumes/RAID/var/tmp/mprsyncup
 PREFIX=${ROOT}/opt/local
-GITROOT=/var/tmp/macports
+GITROOT=${ROOT}
 TBASE=${GITROOT}/trunk/base
 RBASE=${GITROOT}/release/base
 PORTS=${GITROOT}/release/ports
 CONTRIB=${GITROOT}/release/contrib
-RSYNCROOT=/rsync/macports
-MPTOOLSPREFIX=/opt/local
+RSYNCROOT=/Volumes/RAID@PREFIX@/var/rsync/macports
+MPTOOLSPREFIX=@PREFIX@
 MPBIN=${MPTOOLSPREFIX}/bin
 
 # Commands we need. For options to be substituted correctly, these must
@@ -74,8 +74,7 @@
 RELEASE_URL_FILE=config/RELEASE_URL
 
 # private key to use for signing
-# XXX set real path
-PRIVKEY=""
+PRIVKEY="@PREFIX@/var/keys/macports-privkey.pem"
 
 #
 # Functions
@@ -100,6 +99,14 @@
 }
 
 #
+# Wait for rsync root to exist (may be on a filesystem still being mounted)
+#
+
+while [ ! -d "${RSYNCROOT}" ]; do
+    sleep 1
+done
+
+#
 # Update trunk/base
 #
 
@@ -274,4 +281,3 @@
     ${RM} -rf "${ROOT}"/PortIndex_*
 fi
 
-printf "Done\n"
