--- mprsyncup.orig	2017-10-06 11:47:02.000000000 -0500
+++ mprsyncup	2017-10-30 11:41:43.000000000 -0500
@@ -28,34 +28,33 @@
 ####
 
 set -e
-set -x
 
 # Commands we need. For options to be substituted correctly, these must
 # not be substituted within double quotes. Thus, there must not be any
 # globbing characters, and the command itself must not contain spaces.
-GIT="/opt/local/bin/git"
-RSYNC="/opt/local/bin/rsync -q"
+GIT="@PREFIX@/bin/git"
+RSYNC="@PREFIX@/bin/rsync -q"
 RM="/bin/rm"
 MKDIR="/bin/mkdir"
-MV="/opt/local/bin/gmv"
+MV="@PREFIX@/bin/gmv"
 LN="/bin/ln"
 TAR="/usr/bin/tar"
 OPENSSL="/usr/bin/openssl"
 AWK="/usr/bin/awk"
-STAT="/opt/local/bin/gstat"
+STAT="@PREFIX@/bin/gstat"
 BASENAME="/usr/bin/basename"
 
 # Paths we'll work on:
-ROOT=/var/tmp/macports
+ROOT=/Volumes/RAID/var/tmp/mprsyncup
 PREFIX=${ROOT}/opt/local
-GITROOT=/var/tmp/macports
+GITROOT=${ROOT}
 TBASE=${GITROOT}/trunk/base
 RBASE=${GITROOT}/release/base
 PORTS=${GITROOT}/release/ports
-RSYNCROOT=/rsync/macports
+RSYNCROOT=/Volumes/RAID@PREFIX@/var/rsync/macports
 PORTINDEX=${PREFIX}/bin/portindex
 
-PATH=${PREFIX}/bin:/bin:/usr/bin:/usr/sbin:/opt/local/bin
+PATH=${PREFIX}/bin:/bin:/usr/bin:/usr/sbin:@PREFIX@/bin
 
 # Platforms we generate indexes for. This is intentionally split on
 # whitespace later.
@@ -67,8 +66,7 @@
 RELEASE_URL_FILE=config/RELEASE_URL
 
 # private key to use for signing
-# XXX set real path
-PRIVKEY=""
+PRIVKEY="@PREFIX@/var/keys/macports-privkey.pem"
 
 #
 # Functions
@@ -93,6 +91,14 @@
 }
 
 #
+# Wait for rsync root to exist (may be on a filesystem still being mounted)
+#
+
+while [ ! -d "${RSYNCROOT}" ]; do
+    sleep 1
+done
+
+#
 # Update trunk/base
 #
 
@@ -239,3 +245,5 @@
     done
     ${RM} -rf "${ROOT}"/PortIndex_*
 fi
+
+printf "Done\n"
