--- mprsyncup.orig	2016-11-16 02:46:17.000000000 -0600
+++ mprsyncup	2016-11-16 02:51:46.000000000 -0600
@@ -29,31 +29,30 @@
 ####
 
 set -e
-set -x
 
 # Commands we need:
-SVN="/opt/local/bin/svn --non-interactive"
-RSYNC="/opt/local/bin/rsync -q"
+SVN="@PREFIX@/bin/svn --non-interactive"
+RSYNC="@PREFIX@/bin/rsync -q"
 RM="/bin/rm"
 MKDIR="/bin/mkdir"
-MV="/opt/local/bin/gmv"
+MV="@PREFIX@/bin/gmv"
 LN="/bin/ln"
 TAR="/usr/bin/tar"
 OPENSSL="/usr/bin/openssl"
 SED="/usr/bin/sed"
-STAT="/opt/local/bin/gstat"
+STAT="@PREFIX@/bin/gstat"
 
 # Paths we'll work on:
-ROOT=/var/tmp/macports
+ROOT=/var/tmp/mprsyncup
 PREFIX=${ROOT}/opt/local
-SVNROOT=/var/tmp/macports
+SVNROOT=${ROOT}
 TBASE=${SVNROOT}/trunk/base
 RBASE=${SVNROOT}/release/base
 PORTS=${SVNROOT}/release/ports
-RSYNCROOT=/rsync/macports
+RSYNCROOT=@PREFIX@/var/rsync/macports
 PORTINDEX=${PREFIX}/bin/portindex
 
-PATH=${PREFIX}/bin:/bin:/usr/bin:/usr/sbin:/opt/local/bin
+PATH=${PREFIX}/bin:/bin:/usr/bin:/usr/sbin:@PREFIX@/bin
 
 # platforms we generate indexes for
 PLATFORMS="8_powerpc 8_i386 9_powerpc 9_i386 10_i386 11_i386 12_i386 13_i386 14_i386 15_i386 16_i386"
@@ -64,8 +63,7 @@
 RELEASE_URL_FILE=config/RELEASE_URL
 
 # private key to use for signing
-# XXX set real path
-PRIVKEY=""
+PRIVKEY="@PREFIX@/var/keys/macports-privkey.pem"
 
 #
 # Functions
@@ -239,3 +237,5 @@
     done
     ${RM} -rf "${ROOT}"/PortIndex_*
 fi
+
+printf "Done\n"
