--- mprsyncup.orig	2019-06-29 18:16:49.000000000 -0500
+++ mprsyncup	2019-06-29 18:55:20.000000000 -0500
@@ -31,15 +31,15 @@
 set -x
 
 # Paths we'll work on:
-ROOT=/var/tmp/macports
+ROOT=/Volumes/RAID/var/tmp/mprsyncup
 PREFIX=${ROOT}/opt/local
-GITROOT=/var/tmp/macports
+GITROOT=${ROOT}
 TBASE=${GITROOT}/trunk/base
 RBASE=${GITROOT}/release/base
 PORTS=${GITROOT}/release/ports
 CONTRIB=${GITROOT}/release/contrib
-RSYNCROOT=/rsync/macports
-MPTOOLSPREFIX=/opt/local
+RSYNCROOT=/Volumes/RAID@PREFIX@/var/rsync/macports
+MPTOOLSPREFIX=@PREFIX@
 MPBIN=${MPTOOLSPREFIX}/bin
 
 # Commands we need. For options to be substituted correctly, these must
@@ -73,8 +73,7 @@
 RELEASE_URL_FILE=config/RELEASE_URL
 
 # private key to use for signing
-# XXX set real path
-PRIVKEY=""
+PRIVKEY="@PREFIX@/var/keys/macports-privkey.pem"
 
 #
 # Functions
@@ -99,6 +98,14 @@
 }
 
 #
+# Wait for rsync root to exist (may be on a filesystem still being mounted)
+#
+
+while [ ! -d "${RSYNCROOT}" ]; do
+    sleep 1
+done
+
+#
 # Update trunk/base
 #
 
@@ -264,3 +271,5 @@
     done
     ${RM} -rf "${ROOT}"/PortIndex_*
 fi
+
+printf "Done\n"
