--- master.cfg.orig	2016-08-05 09:34:53.000000000 -0500
+++ master.cfg	2016-08-05 10:01:13.000000000 -0500
@@ -23,13 +23,13 @@
 privkey      = ''
 slaveport    = 9989
 httpport     = 8010
-buildboturl  = "http://localhost:8010/"
+buildboturl  = "https://build-test.macports.org/"
 htpasswdfile = "htpasswd"
 mpbbsvnurl   = "https://svn.macports.org/repository/macports/contrib/mp-buildbot"
 svnurl       = "https://svn.macports.org/repository/macports/trunk"
 archivesite  = "https://packages.macports.org"
 slaveprefix  = "/opt/local"
-toolsprefix  = "/opt/mports"
+toolsprefix  = "/opt/bblocal"
 
 if os.path.exists(_path('config.json')):
     with open(_path('config.json')) as f:
@@ -242,7 +242,7 @@
 if production:
     prefix='/opt/local'
     dlhost='packages@packages-origin.macports.org'
-    dlpath='/var/www/html/packages'
+    dlpath='@PREFIX@/var/rsync/macports/packages'
 else:
     prefix=slaveprefix
     dlhost=''
