--- master.cfg.orig	2018-03-08 09:54:55.000000000 -0600
+++ master.cfg	2018-03-08 09:56:44.000000000 -0600
@@ -422,8 +422,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix = '/opt/local'
-    dlhost = 'packages@packages-origin.macports.org'
-    dlpath = '/var/www/html/packages'
+    dlhost = ''
+    dlpath = '/Volumes/RAID@PREFIX@/var/rsync/macports/packages'
 else:
     prefix = config['slaveprefix']
     dlhost = ''
