--- master.cfg.orig	2016-07-26 22:39:10.000000000 -0500
+++ master.cfg	2016-08-03 12:59:25.000000000 -0500
@@ -23,13 +23,13 @@
 privkey      = ''
 slaveport    = 9989
 httpport     = 8010
-buildboturl  = "http://localhost:8010/"
+buildboturl  = "./"
 htpasswdfile = "htpasswd"
 mpbbsvnurl   = "https://svn.macports.org/repository/macports/contrib/mp-buildbot"
 svnurl       = "https://svn.macports.org/repository/macports/trunk"
 archivesite  = "https://packages.macports.org"
 slaveprefix  = "/opt/local"
-toolsprefix  = "/opt/mports"
+toolsprefix  = "/opt/bblocal"
 
 if os.path.exists(_path('config.json')):
     with open(_path('config.json')) as f:
@@ -40,7 +40,7 @@
             locals[key] = configdata[key]
 
 # TODO: the following list should be configured in slaves.json
-build_platforms = ["10.6-x86_64"]
+build_platforms = ["10.6-x86_64", "10.6-x86_64-legacy", "10.6-i386", "10.6-i386-legacy", "10.7-x86_64", "10.7-x86_64-legacy", "10.8-x86_64", "10.8-x86_64-legacy", "10.9-x86_64", "10.11-x86_64"]
 #build_platforms = ["snowleopard-x86_64", "lion-x86_64", "mtln-x86_64", "mavericks-x86_64", "yosemite-x86_64", "elcapitan-x86_64", "oelinux-6.3-x86_64"]
 
 # Allow spaces and tabs in property values
@@ -117,7 +117,7 @@
 portsfilter = ChangeFilter(filter_fn=change_has_ports)
 basefilter = ChangeFilter(filter_fn=change_has_base)
 
-base_buildernames = ["base-"+plat for plat in build_platforms]
+base_buildernames = ["base-"+plat for plat in build_platforms if 'legacy' not in plat and '10.6-i386' not in plat]
 portwatcher_buildernames = ["portwatcher-"+plat for plat in build_platforms if 'linux' not in plat]
 portbuilder_buildernames = ["portbuilder-"+plat for plat in build_platforms if 'linux' not in plat]
 portbuilder_triggerables = ["porttrigger-"+plat for plat in build_platforms if 'linux' not in plat]
@@ -141,14 +141,14 @@
     ForceScheduler(
         name="base_force",
         builderNames=base_buildernames),
-    ForceScheduler(
-        name="portbuilder_force",
-        builderNames=portbuilder_buildernames,
-        properties=[StringParameter(
-            name="portname",
-            label="Port name:",
-            default="")
-        ]),
+#    ForceScheduler(
+#        name="portbuilder_force",
+#        builderNames=portbuilder_buildernames,
+#        properties=[StringParameter(
+#            name="portname",
+#            label="Port name:",
+#            default="")
+#        ]),
     ForceScheduler(
         name="portwatcher_force",
         builderNames=portwatcher_buildernames,
@@ -241,7 +241,7 @@
 if production:
     prefix='/opt/local'
     dlhost='packages@packages-origin.macports.org'
-    dlpath='/var/www/html/packages'
+    dlpath='@PREFIX@/var/rsync/macports/packages'
 else:
     prefix=slaveprefix
     dlhost=''
@@ -395,16 +395,10 @@
 # XXX: slavenames assignment should be automatic and more generic
 portsslaves = {}
 baseslaves = {}
-if production:
-    slavenames = slavedata.keys()
-    for plat in build_platforms:
-        baseslaves[plat]  = filter(lambda x: x.endswith(plat+"-base"),  slavenames)
-        portsslaves[plat] = filter(lambda x: x.endswith(plat+"-ports"), slavenames)
-#else:
-    #slavenames = ["snowleopard-x86_64"]
-    #slavenames = ["base-10.6-x86_64","10.6-x86_64","10.9-x86_64"]
-    #portsslaves = {build_platforms[0]:slavenames[0],build_platforms[1]:slavenames[1]}
-    #baseslaves = {build_platforms[0]:slavenames[0],build_platforms[1]:slavenames[1]}
+slavenames = slavedata.keys()
+for plat in build_platforms:
+    baseslaves[plat]  = filter(lambda x: x.endswith(plat+"-base"),  slavenames)
+    portsslaves[plat] = filter(lambda x: x.endswith(plat+"-ports"), slavenames)
 
 c['builders']=[]
 for plat in build_platforms:
@@ -412,6 +406,7 @@
         name="base-" + plat,
         slavenames=["base-" + plat],
         factory=base_factory,
+        tags=["base", plat],
         env={"PATH": "/usr/bin:/bin:/usr/sbin:/sbin"})]
     if 'linux' not in plat:
         c['builders'] += [
@@ -419,11 +414,13 @@
                 name="portwatcher-" + plat,
                 slavenames=["ports-" + plat],
                 factory=make_portwatcher_factory("porttrigger-" + plat),
+                tags=["ports", plat],
                 env={"PATH": "/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin"}),
             BuilderConfig(
                 name="portbuilder-" + plat,
                 slavenames=["ports-" + plat],
                 factory=portbuilder_factory,
+                tags=["ports", plat],
                 env={"PATH": toolsprefix + "/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin"})]
 
 ####### STATUS TARGETS
