--- master.cfg.orig	2021-03-20 01:01:54.000000000 -0500
+++ master.cfg	2021-03-20 01:06:14.000000000 -0500
@@ -444,10 +444,10 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix = '/opt/local'
-    dlhost = 'packages@packages-origin.macports.org'
+    dlhost = ''
     dlhost_private = dlhost
-    dlpath = '/var/www/html/packages'
-    dlpath_private = '/var/www/html/packages-private'
+    dlpath = '/Volumes/RAID@PREFIX@/var/rsync/macports/packages'
+    dlpath_private = '/Volumes/RAID@PREFIX@/var/www/macports/packages-private'
 else:
     prefix = config['slaveprefix']
     dlhost = ''
@@ -863,6 +863,7 @@
         haltOnFailure=True))
     jobs_man_factory.addStep(steps.Configure(
         command="./standard_configure.sh",
+        env={'UNIVERSAL': 'no'},
         logfiles={'config.log': 'config.log'},
         haltOnFailure=True))
     jobs_man_factory.addStep(steps.Compile(
