Index: master.cfg
===================================================================
--- master.cfg	(revision 153038)
+++ master.cfg	(working copy)
@@ -91,7 +91,8 @@
 # poller is used for local testing but PBChangeSource (which relies on
 # notifications from a post-commit script) should be used in production
 
-if config['production']:
+#if config['production']:
+if False:
     # TODO
     from buildbot.changes.pb import PBChangeSource
     sourcedata = []
@@ -102,7 +103,7 @@
     from buildbot.changes.svnpoller import SVNPoller
     c['change_source'] = (SVNPoller(
         svnurl='https://svn.macports.org/repository/macports/trunk',
-        #svnbin='/opt/local/bin/svn',
+        svnbin='@PREFIX@/bin/svn',
         pollinterval=300,
         category='macports',
         project='ports'))
@@ -294,8 +295,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix='/opt/local'
-    dlhost='packages@packages-origin.macports.org'
-    dlpath='/var/www/html/packages'
+    dlhost=''
+    dlpath='@PREFIX@/var/rsync/macports/packages'
 else:
     prefix=config['slaveprefix']
     dlhost=''
@@ -608,11 +609,9 @@
 from buildbot.status import html
 from buildbot.status.web import auth, authz
 
-from buildbot.status.web.auth import HTPasswdAuth
+from buildbot.status.web.auth import HTPasswdAprAuth
 
-# add new users with crypt, not md5/sha1:
-# htpasswd -d -n $USER
-htauth = HTPasswdAuth(config['htpasswdfile'])
+htauth = HTPasswdAprAuth(config['htpasswdfile'])
 
 authz_cfg=authz.Authz(
     auth               = htauth,
@@ -747,8 +746,6 @@
 
         if len(maintainers_to_notify) > 0:
             text.append("\nResponsible maintainers:\n\t- {}".format("\n\t- ".join(sorted(maintainers_to_notify))))
-            for user in maintainers_to_notify:
-                interested_users.add(user)
 
         # links to individual builds
         text.append("\nLinks to individual build jobs:")
