--- master.cfg.orig	2018-04-11 22:02:30.000000000 -0500
+++ master.cfg	2018-04-11 22:04:27.000000000 -0500
@@ -442,8 +442,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix = '/opt/local'
-    dlhost = 'packages@packages-origin.macports.org'
-    dlpath = '/var/www/html/packages'
+    dlhost = ''
+    dlpath = '/Volumes/RAID@PREFIX@/var/rsync/macports/packages'
 else:
     prefix = config['slaveprefix']
     dlhost = ''
