--- master.cfg.orig	2026-01-20 01:12:28.000000000 -0600
+++ master.cfg	2026-01-20 01:16:25.000000000 -0600
@@ -455,12 +455,12 @@
     prefix = config['slaveprefix']
 
 dlhost = config.get('dlhost',
-    'packages@packages-origin.macports.org' if config['production'] else '')
+    '' if config['production'] else '')
 dlhost_private = config.get('dlhost_private', dlhost)
 dlpath = config.get('dlpath',
-    '/var/www/html/packages' if config['production'] else './deployed_archives')
+    '/Volumes/RAID@PREFIX@/var/rsync/macports/packages' if config['production'] else './deployed_archives')
 dlpath_private = config.get('dlpath_private',
-    '/var/www/html/packages-private' if config['production'] else './deployed_archives_private')
+    '/Volumes/RAID@PREFIX@/var/www/macports/packages-private' if config['production'] else './deployed_archives_private')
 
 # Path to upload archives from on the builder.
 ulpath = config.get('ulpath',  'archive_staging')
@@ -895,6 +895,7 @@
         haltOnFailure=True))
     jobs_man_factory.addStep(steps.Configure(
         command="./standard_configure.sh",
+        env={'UNIVERSAL': 'no'},
         logfiles={'config.log': 'config.log'},
         haltOnFailure=True))
     jobs_man_factory.addStep(steps.Compile(
