Index: master.cfg
===================================================================
--- master.cfg	(revision 152651)
+++ master.cfg	(working copy)
@@ -91,7 +91,8 @@
 # poller is used for local testing but PBChangeSource (which relies on
 # notifications from a post-commit script) should be used in production
 
-if config['production']:
+#if config['production']:
+if False:
     # TODO
     from buildbot.changes.pb import PBChangeSource
     sourcedata = []
@@ -102,7 +103,7 @@
     from buildbot.changes.svnpoller import SVNPoller
     c['change_source'] = (SVNPoller(
         svnurl='https://svn.macports.org/repository/macports/trunk',
-        #svnbin='/opt/local/bin/svn',
+        svnbin='@PREFIX@/bin/svn',
         pollinterval=300,
         category='macports',
         project='ports'))
@@ -294,8 +295,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix='/opt/local'
-    dlhost='packages@packages-origin.macports.org'
-    dlpath='/var/www/html/packages'
+    dlhost=''
+    dlpath='@PREFIX@/var/rsync/macports/packages'
 else:
     prefix=config['slaveprefix']
     dlhost=''
@@ -747,8 +748,6 @@
 
         if len(maintainers_to_notify) > 0:
             text.append("\nResponsible maintainers:\n\t- {}".format("\n\t- ".join(sorted(maintainers_to_notify))))
-            for user in maintainers_to_notify:
-                interested_users.add(user)
 
         # links to individual builds
         text.append("\nLinks to individual build jobs:")
@@ -777,26 +776,26 @@
 if config['production']:
     # send mail about base failures to users on the blamelist
     mn = MailNotifier(
-        fromaddr="buildbot@macports.org",
-        extraHeaders={"Reply-To": "noreply@macports.org"},
+        fromaddr="buildbot@ryandesign.com",
+        extraHeaders={"Reply-To": "noreply@ryandesign.com"},
         # unless lookup is defined, users have to be configured locally
         # maybe a smarter function is needed, but lookup="" does it for now
         lookup="",
         mode=("problem"),
         builders=base_buildernames,
-        #extraRecipients=["..."],
+        extraRecipients=["macportsbuild@ryandesign.com"],
         #smtpPort=25,
         #relayhost="localhost",
         sendToInterestedUsers=True)
     c['status'].append(mn)
 
     mn = PortsMailNotifier(
-        fromaddr="buildbot@macports.org",
-        extraHeaders={"Reply-To": "noreply@macports.org"},
+        fromaddr="buildbot@ryandesign.com",
+        extraHeaders={"Reply-To": "noreply@ryandesign.com"},
         lookup="",
         mode=("failing"),
         builders=portwatcher_buildernames,
-        #extraRecipients=["..."],
+        extraRecipients=["macportsbuild@ryandesign.com"],
         #smtpPort=25,
         #relayhost="localhost",
         sendToInterestedUsers=True,
@@ -805,12 +804,12 @@
 
     # notifications about exceptions
     mn = MailNotifier(
-        fromaddr="buildbot@macports.org",
-        extraHeaders={"Reply-To": "noreply@macports.org"},
+        fromaddr="buildbot@ryandesign.com",
+        extraHeaders={"Reply-To": "noreply@ryandesign.com"},
         mode=("exception"),
-        extraRecipients=["admin@macports.org"],
+        extraRecipients=["macportsbuild@ryandesign.com"],
         sendToInterestedUsers=False)
-    # c['status'].append(mn)
+    c['status'].append(mn)
 
 ####### PROJECT IDENTITY
 
