--- master.cfg.orig	2018-03-07 11:17:06.000000000 -0600
+++ master.cfg	2018-03-07 11:27:37.000000000 -0600
@@ -383,8 +383,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix = '/opt/local'
-    dlhost = 'packages@packages-origin.macports.org'
-    dlpath = '/var/www/html/packages'
+    dlhost = ''
+    dlpath = '/Volumes/RAID@PREFIX@/var/rsync/macports/packages'
 else:
     prefix = config['slaveprefix']
     dlhost = ''
