--- master.cfg.orig	2018-08-28 18:56:28.000000000 -0500
+++ master.cfg	2018-09-30 05:10:04.000000000 -0500
@@ -442,8 +442,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix = '/opt/local'
-    dlhost = 'packages@packages-origin.macports.org'
-    dlpath = '/var/www/html/packages'
+    dlhost = ''
+    dlpath = '/Volumes/RAID@PREFIX@/var/rsync/macports/packages'
 else:
     prefix = config['slaveprefix']
     dlhost = ''
@@ -839,6 +839,7 @@
         haltOnFailure=True))
     jobs_man_factory.addStep(steps.Configure(
         command="./standard_configure.sh",
+        env={'UNIVERSAL': 'no'},
         logfiles={'config.log': 'config.log'},
         haltOnFailure=True))
     jobs_man_factory.addStep(steps.Compile(
