Index: master.cfg
===================================================================
--- master.cfg	(revision 152525)
+++ master.cfg	(working copy)
@@ -84,7 +84,8 @@
 # poller is used for local testing but PBChangeSource (which relies on
 # notifications from a post-commit script) should be used in production
 
-if config['production']:
+#if config['production']:
+if False:
     # TODO
     from buildbot.changes.pb import PBChangeSource
     sourcedata = []
@@ -95,7 +96,7 @@
     from buildbot.changes.svnpoller import SVNPoller
     c['change_source'] = (SVNPoller(
         svnurl='https://svn.macports.org/repository/macports/trunk',
-        #svnbin='/opt/local/bin/svn',
+        svnbin='@PREFIX@/bin/svn',
         pollinterval=300,
         category='macports',
         project='ports'))
@@ -279,8 +280,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix='/opt/local'
-    dlhost='packages@packages-origin.macports.org'
-    dlpath='/var/www/html/packages'
+    dlhost=''
+    dlpath='@PREFIX@/var/rsync/macports/packages'
 else:
     prefix=config['slaveprefix']
     dlhost=''
@@ -320,6 +321,7 @@
 
 def make_portwatcher_factory(triggerable):
     portwatcher_factory = BuildFactory()
+    portwatcher_factory.useProgress = False
     portwatcher_factory.workdir = '../build'
 
     # get mp-buildbot; we'll do the checkout of base and dports via these scripts
@@ -395,6 +397,7 @@
 # -- Port Builder --
 
 portbuilder_factory = BuildFactory()
+portbuilder_factory.useProgress = False
 portbuilder_factory.workdir = '../build'
 logdir = os.path.join(portbuilder_factory.workdir, "logs")
 
@@ -701,7 +704,8 @@
         'subject' : subject
     }
 
-if config['production']:
+#if config['production']:
+if False:
     # send mail about base failures to users on the blamelist
     mn = MailNotifier(
         fromaddr="noreply@macports.org",
