--- master.cfg.orig	2016-10-28 21:40:53.000000000 -0500
+++ master.cfg	2016-10-28 21:42:03.000000000 -0500
@@ -312,8 +313,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix = '/opt/local'
-    dlhost = 'packages@packages-origin.macports.org'
-    dlpath = '/var/www/html/packages'
+    dlhost = ''
+    dlpath = '@PREFIX@/var/rsync/macports/packages'
 else:
     prefix = config['slaveprefix']
     dlhost = ''
