--- master.cfg.orig	2017-05-01 20:27:51.000000000 -0500
+++ master.cfg	2017-05-01 20:27:52.000000000 -0500
@@ -360,8 +360,8 @@
 # because archives must be built with prefix=/opt/local
 if config['production']:
     prefix = '/opt/local'
-    dlhost = 'packages@packages-origin.macports.org'
-    dlpath = '/var/www/html/packages'
+    dlhost = ''
+    dlpath = '/Volumes/RAID@PREFIX@/var/rsync/macports/packages'
 else:
     prefix = config['slaveprefix']
     dlhost = ''
